### PR TITLE
Cleanbots with knives now also stab when moving into your tile, +refactors and other fun changes

### PR DESCRIPTION
**Pull request by Ryll-Ryll**
## About The Pull Request
This is a long overdue follow up to #47590. Attaching a knife to a cleanbot usually ends up getting the cleanbot instantly murdered since they only trigger stabbing if you step onto them, so all you have to do to avoid them is stand still. Now, cleanbots will also stab people when they move into the person's tile, making them much more proactive. Hilariously enough, since it's possible for their stabs to give you cuts that bleed, and cleanbots like to clean blood, it's perfectly possible for the cleanbot to "chase" after you as you run away! Cleanbots won't stab you on moving into your tile if you're laying down or buckled.

Armed cleanbots with their safeties disabled will now have a chance to block or even parry attacks made against them by bleeding crewmembers that scales with how much you're bleeding (the cleanbot tries to clean up your blood and accidentally blocks/parries you with its weapon as a result)! Attacking it with a weapon is more likely to block the attack than to parry and attack back, but trying to punch the cleanbot with your bare hands is much more likely to proc a response, which will always be a parry. If you're not bleeding though, you won't have anything to worry about!

https://streamable.com/csh1i5

Other changes in no particular order:

- Items being attachable to cleanbots is now a trait, TRAIT_CLEANBOT_COMPATIBLE, which has been added to forks and scalpels! The admin-only super pointy tape also grants this trait, so you can use it to do fun things like attach the captain's saber to a cleanbot!
- Added a suffix for stabbing paramedics, E.M.T.
- Cleanbots will no longer have their initial names reset when gaining new titles

## Why It's Good For The Game
Knifebots are funny and all, but they're incredibly incredibly fragile and easy to defeat. They're still pretty easy to kill and mostly just minor nuisances, but they require a bit more forethought to kill now in melee.

## Changelog

:cl: Ryll/Shaps
balance: Cleanbots with knives taped to them can now stab people by moving into them, instead of only being able to stab people who step onto them! They will ignore people who are buckled or laying prone, however.
expansion: Cleanbots with knives who have their safeties disabled can now block or even parry attacks made on them by bleeding crewmembers, as the Cleanbot tries to clean up their gushing blood, and accidentally fighting back with their weapon along the way! The defense chance scales with how much the attacker is bleeding, and is much more likely to happen when the attacker uses their fists!
expansion: You can now attach forks and scalpels to cleanbots.
fix: Cleanbots should now retain their original name when they steal peoples' titles

/:cl:

### CODE REVIEW

**Review by tralezab — CHANGES_REQUESTED** [review]
love the emag defines

File: code/modules/mob/living/simple_animal/bot/cleanbot.dm [hunk 1]
```diff
@@ -33,36 +46,60 @@
...
 	var/list/stolen_valor
 
 	var/static/list/officers = list("Captain", "Head of Personnel", "Head of Security")
 	var/static/list/command = list("Captain" = "Cpt.","Head of Personnel" = "Lt.")
 	var/static/list/security = list("Head of Security" = "Maj.", "Warden" = "Sgt.", "Detective" =  "Det.", "Security Officer" = "Officer")
 	var/static/list/engineering = list("Chief Engineer" = "Chief Engineer", "Station Engineer" = "Engineer", "Atmospherics Technician" = "Technician")
-	var/static/list/medical = list("Chief Medical Officer" = "C.M.O.", "Medical Doctor" = "M.D.", "Chemist" = "Pharm.D.")
+	var/static/list/medical = list("Chief Medical Officer" = "C.M.O.", "Medical Doctor" = "M.D.", "Chemist" = "Pharm.D.", "Paramedic" = "E.M.T.")
 	var/static/list/research = list("Research Director" = "Ph.D.", "Roboticist" = "M.S.", "Scientist" = "B.S.")
 	var/static/list/legal = list("Lawyer" = "Esq.")
```
[the remark this answers is not in the document]
Comment (Ryll-Ryll) [reply code/modules/mob/living/simple_animal/bot/cleanbot.dm:64]: added psychologist, but service roles don't really have ranks so

File: code/modules/mob/living/simple_animal/bot/cleanbot.dm [hunk 2]
```diff
@@ -176,21 +243,56 @@
...
+			else // otherwise it just blocks the attack
+				visible_message(span_danger("[src] spins around trying to clean the blood coming from [user], accidentally parrying [user.p_their()] attack perfectly with \the [taped_weapon]!"),
+									span_userdanger("Trying to clean the blood flowing from [user], you accidentally parry [user.p_their()] attack perfectly!"), COMBAT_MESSAGE_RANGE, ignored_mobs = user)
+				to_chat(user, span_userdanger("You try to strike [src] with \the [W], but it accidentally blocks your attack while trying to clean your flowing blood!"))
+			return
+
+	return ..()
+
+/mob/living/simple_animal/bot/cleanbot/attack_hand(mob/living/carbon/human/user, list/modifiers)
+	if(!(taped_weapon && user.combat_mode && emagged == BOT_EMAGGED_OVERDRIVE))
```
[the remark this answers is not in the document]
Comment (Ryll-Ryll) [reply code/modules/mob/living/simple_animal/bot/cleanbot.dm:275]: eh, that can be changed later if needed

**Comment by Ryll-Ryll** [comment]
This should be good to merge by the way

**Pull request closed without merging**